sympy/stats/rv.py

```diff
@@ -121,6 +121,27 @@ def as_boolean(self):
         return And(self.fulldomain.as_boolean(), self.condition)
 
 
+class RandomDistribution(Basic):
+    def __call__(self, *args):
+        return self.pdf(*args)
+
+    def handle_compound_dist(self, expr):
+        from sympy.stats.rv import density
+        from sympy.concrete.summations import summation
+        for arg in self.args:
+            if isinstance(arg, RandomSymbol):
+                sym, dom = arg.symbol, arg.pspace.set
+                lpdf = density(arg)(sym)
+                expr = expr.replace(arg, sym)*lpdf
+                if arg.pspace.is_Continuous:
+                    expr = integrate(expr, (sym, dom))
```

Comment:
`integrate` does not currently support the set notation, but it should in the future. Better leave a note.